application/src/main/java/org/thingsboard/server/service/entitiy/entityView/DefaultTbEntityViewService.java

```diff
@@ -214,6 +227,35 @@ public EntityView unassignEntityViewFromCustomer(TenantId tenantId, EntityViewId
         }
     }
 
+    @Override
+    public ListenableFuture<List<EntityView>> findEntityViewsByTenantIdAndEntityIdAsync(TenantId tenantId, EntityId entityId) {
+        Map<EntityId, List<EntityView>> localCacheByTenant = localCache.computeIfAbsent(tenantId, (k) -> new ConcurrentReferenceHashMap<>());
+        List<EntityView> fromLocalCache = localCacheByTenant.get(entityId);
+        if (fromLocalCache != null) {
+            return Futures.immediateFuture(fromLocalCache);
+        }
+
+        ListenableFuture<List<EntityView>> future = entityViewService.findEntityViewsByTenantIdAndEntityIdAsync(tenantId, entityId);
+
+        return Futures.transform(future, (entityViewList) -> {
+            localCacheByTenant.put(entityId, entityViewList);
+            return entityViewList;
+        }, MoreExecutors.directExecutor());
+    }
+
+    @Override
+    public void onComponentLifecycleMsg(ComponentLifecycleMsg componentLifecycleMsg) {
```

Comment:
please replace with onEntityViewUpdate

Reply:
I would like to set a generic Event sourcing interface for any entity. The idea is to not stick to a particular entity name. I agree that refactoring is required.

This extracted to separate PR https://github.com/thingsboard/thingsboard/pull/6785